### DISCUSSION TITLE
Add Icon

### DISCUSSION THREAD

**Discussion by hirockstar** [j471n/j471n.in#20] [Q&A] [answered]
bro i want to add icons to topbar links how can i do that.
https://github.com/j471n/j471n.in/blob/ab4d956a61392eb46e630aa65e99b21c97f0f96e/utils/utils.ts#L5

we want to add icon like this 
![Screenshot 2023-02-12 164741](https://user-images.githubusercontent.com/45270927/218307846-1f5bc8e5-dc1d-4bc6-b955-b56ac8fb853c.png)

**Comment by j471n** [comment]
You can do either of the following things:
1. You can customize the `navigationRoutes` make it an array of objects with `route`, `Icon` and some other properties if you want.
2. You can leave it as it is. Just add the switch case or a function that will give you the Icon based on route.

```jsx

function getIcon(route) {
  switch(route) {
   case "/about":
     return <About />
  }
}

```


Something like that.

**Reply by hirockstar** [reply]
can you specify more bcz i do not understand?

**Reply by j471n** [reply, marked as answer]
You can do two things -

## FIRST

```js
const navigationRoutes = [
  {
    route: "/",
    name:"Home",
    Icon: <HomeIcon />
  },
//........more routes
// something like that
]
```

Then import this in your navbar and use it. 


```js
navigationRoutes.map(nav => (
   <Link href={nav.route}>
      {nav.Icon}
     <p>{nav.name}</p>
   </Link>
))
```


## SECOND

```js
const navigationRoutes = [
  "home",
  "about",
  "blog",
]
```

now what you can do is create a function that will return the Icon based on the navigation route name such as following:

```js
// import icons...........

export function getIcon(route) {
  switch(route) {
   case "home"
     return <HomeIcon />
   case "about":
     return <AboutIcon />
   case "about":
     return <BlogIcon />
  }
}

// or the following one



export function getIcon(route) {
  switch(route) {
   case "home"
     return HomeIcon;
   case "about":
     return AboutIcon;
   case "about":
     return BlogIcon;
  }
}
```


Now then you can use it as following:

```js
import { getIcon } from ".....path"

navigationRoutes.map(route=> {
  
   const Icon = getIcon(route);

   return (<Link href={`/${route}`}>
      {Icon} // <Icon /> (for second method)
     <p>{nav}</p>
   </Link>
  )
})
```

Try any of these. 

For more info look at this [code](https://github.com/j471n/j471n.in/blob/main/components/MDXComponents/CodeTitle.tsx).
